src/equations/2d/linear_scalar_advection.jl

```diff
@@ -24,18 +24,27 @@ get_name(::LinearScalarAdvectionEquation2D) = "LinearScalarAdvectionEquation2D"
 varnames_cons(::LinearScalarAdvectionEquation2D) = SVector("scalar")
 varnames_prim(::LinearScalarAdvectionEquation2D) = SVector("scalar")
 
+function x_trans_periodic(x)
+  #Fixed domain_length and center.
+  domain_length = [2,2]
+  center = [0,0]
+  x = (x - center) .% (domain_length)
+  x += ((x .<= -0.5*domain_length) .- (x .>= 0.5*domain_length)) .* (domain_length)
+  return x + center
+end
 
 # Set initial conditions at physical location `x` for time `t`
 function initial_conditions_gauss(x, t, equation::LinearScalarAdvectionEquation2D)
   # Store translated coordinate for easy use of exact solution
-  x_trans = x - equation.advectionvelocity * t
+
+  x_trans = x_trans_periodic(x - equation.advectionvelocity * t)
 
   return @SVector [exp(-(x_trans[1]^2 + x_trans[2]^2))]
 end
 
 function initial_conditions_convergence_test(x, t, equation::LinearScalarAdvectionEquation2D)
   # Store translated coordinate for easy use of exact solution
-  x_trans = x - equation.advectionvelocity * t
+  x_trans = x_trans_periodic(x - equation.advectionvelocity * t)
```

Comment:
Here it is not necessary, since the IC (initial condition) is already periodic. Thus to reduce computational time, I suggest to revert it back to the original implementation.
```suggestion
  x_trans = x - equation.advectionvelocity * t
```

Reply:
Thanks for your feedback. I´ll try to adress all of it. 
I´ll undo this change for all initial conditions other than the Gaussian one.